ovs/extensions/support/agent.py

```diff
@@ -143,38 +183,121 @@ def _process_task(self, task_code, metadata):
         finally:
             SupportAgent._logger.debug('Completed')
 
+    def open_tunnel(self, metadata):
+        """
+        Opens up the openvpn tunnel
+        :param metadata: Metadata about the files to use for tunnel
+        :type metadata: dict
+        Example: {'files': {...}}.
+        The files dict will contain file names (keys) and their contents (value) Example: {my_file: my_file_contents}
+        These files will always be /etc/openvpn/ovs_ + 'ca.crt', 'ta.key', '{0}.crt'.format(identifier), '{0}.key'.format(identifier), '{0}.conf'.format(identifier)
+        with identifier = cluster id + node id
+        :return: None
+        :rtype: NoneType
+        """
+        # Close tunnel and update configs
+        self.close_tunnel(metadata)
+        for filename, contents in metadata['files'].iteritems():
+            with open(filename, 'w') as the_file:
+                the_file.write(base64.b64decode(contents))
+        if self._service_type == 'upstart':
+            check_output('service openvpn start', shell=True)
+        else:
+            check_output("systemctl start '{0}'".format(self._openvpn_service_name), shell=True)
+
+    @staticmethod
+    def upload_files(metadata):
+        """
+        Upload the collected log files to a destination
+        :param metadata: Metadata about the destination.
+        Example: {'filename': name to upload the tarball as, 'endpoint': ip of the ftp, 'user': an ftp user, 'password': ftp users password}
+        :type metadata: dict
+        :return: None
+        :rtype: NoneType
+        """
+        logfile = check_output('ovs collect logs', shell=True).strip()
+        check_output("mv '{0}' '/tmp/{1}'; curl -T '/tmp/{1}' 'ftp://{2}' --user '{3}:{4}'; rm -f '{0}' '/tmp/{1}'".format(
+            logfile.replace(r"'", r"'\''"),
+            metadata['filename'].replace(r"'", r"'\''"),
+            metadata['endpoint'].replace(r"'", r"'\''"),
+            metadata['user'].replace(r"'", r"'\''"),
+            metadata['password'].replace(r"'", r"'\''")), shell=True)
+
+    def close_tunnel(self, metadata):
+        """
+        Closes the openvpn tunnel
+        @TODO: only close the ovs remote tunnel on 14.04 (all tunnels are currently closed)
```

Comment:
Don't put TODOs in comment section because most IDEs don't detect this as a real TODO